### PR TITLE
Support for parsing circles & ellipses

### PR DESCRIPTION
**Pull request by samus**
I added basic circle support as I needed it on a project I'm working on.  It's light on error handling and I probably missed something so I'm happy to incorporate any suggestions to make it better.

### CODE REVIEW

**Comment by arielelkin** [comment]
Thanks @samus !

It looks good to me. 

More systematic error handling is on our more general to-do list anyway. 

@fjolnir any thoughts?

File: SVGEngine.mm [hunk 1]
```diff
@@ -226,9 +259,14 @@ @interface SVGAttributeSet () {
         const char * const attrName  = (char *)xmlTextReaderConstName(_xmlReader),
                    * const attrValue = (char *)xmlTextReaderConstValue(_xmlReader);
 
-        if(strcasecmp("style", attrName) == 0)
-            [attrs addEntriesFromDictionary:_SVGParseStyle(@(attrValue))];
-        else if(strcasecmp("transform", attrName) == 0) {
+        if(strcasecmp("style", attrName) == 0){
+            NSMutableDictionary *style = _SVGParseStyle(@(attrValue));
+            //Don't allow overriding of display:none
+            if ([attrs[@"display"] isEqualToString:@"none"] && style[@"display"] != nil) {
```
Comment (fjolnir) [inline SVGEngine.mm]: That second check feels redundant (if the key isn't present, the next line just does nothing if I recall correctly)

Also the string comparison should probably be case insensitive.

File: SVGEngine.mm [hunk 2]
```diff
@@ -214,6 +219,34 @@ @interface SVGAttributeSet () {
...
+CF_RETURNS_RETAINED CGPathRef svgParser::readCircleTag()
+{
+    NSCAssert(strcasecmp((char*)xmlTextReaderConstName(_xmlReader), "circle") == 0,
+              @"Not on a <circle>");
+    CGPoint const center = {
+        readFloatAttribute(@"cx"),     readFloatAttribute(@"cy")
+    };
+    float radius = readFloatAttribute(@"r");
+    CGMutablePathRef circle = CGPathCreateMutable();
+    CGPathAddArc(circle, NULL, center.x, center.y, radius, 0.0, (360 * M_PI), NO);
```
Comment (fjolnir) [inline SVGEngine.mm]: CGPathAddArc takes an angle in radians, not degrees. (2π)

But we could also just use CGPathAddEllipseInRect I think (rx == ry). That way the code for ellipse & circle could be shared.

**Comment by fjolnir** [comment]
Merged via https://github.com/pocketsvg/PocketSVG/pull/76

**Pull request closed without merging**